plugins/imfile/imfile.c

```diff
@@ -2667,46 +2796,268 @@ fen_processEvent(port_event_t* pportEvent, lstn_t *pLstn, int revents)
-
-	DBGPRINTF("imfile: fen_processEvent: associate port for %s\n", pLstn->pfinf->fobj.fo_name);
-	if (port_associate(port, PORT_SOURCE_FILE, (uintptr_t)&pLstn->pfinf->fobj,
+	fobjp->fo_atime = statFile.st_atim;
+	fobjp->fo_mtime = statFile.st_mtim;
+	fobjp->fo_ctime = statFile.st_ctim;
+	if (port_associate(glport, PORT_SOURCE_FILE, (uintptr_t)fobjp,
 				pLstn->pfinf->events, (void *)pLstn) == -1) {
 		/* Add error processing as required, file may have been deleted/moved. */
-		errmsg.LogError(1, RS_RET_SYS_ERR, "imfile: fen_processEvent: Failed to register file "
+		errmsg.LogError(1, RS_RET_SYS_ERR, "fen_processEventFile: Failed to associated port for file "
 			": %s - errno %d\n", fobjp->fo_name, errno);
 		ABORT_FINALIZE(RS_RET_SYS_ERR);
 	} else {
 		/* Port successfull listening now*/
+		DBGPRINTF("fen_processEventFile: associated port for file %s\n", fobjp->fo_name);
 		pLstn->bPortAssociated = 1;
 	}
 finalize_it:
 	RETiRet;
 }
 
+/* function not used yet, will be needed for wildcards later */
+rsRetVal
+fen_DirSearchFiles(lstn_t *pLstn, int dirIdx)
+{
+	struct file_obj *fobjp = NULL;	/* Helper object */
+	int ftIdx;
+	int result;
+	glob_t files;
+	lstn_t *pLstnNew;
+	/* Helper chars */
+	uchar basedir[MAXFNAME];
+	uchar basefilename[MAXFNAME];
+	uchar *file;
+	DEFiRet;
+
+	DBGPRINTF("fen_DirSearchFiles search for dynamic files with pattern '%s' \n", pLstn->pszFileName);
+	result = glob(	(char*)pLstn->pszFileName,
+			GLOB_MARK|GLOB_NOSORT|GLOB_BRACE,
+			NULL, &files);
+	if(result == 0) {
+		DBGPRINTF("fen_DirSearchFiles found %d matches with pattern '%s' \n", files.gl_pathc, pLstn->pszFileName);
+		for(unsigned i = 0 ; i < files.gl_pathc ; i++) {
+			/* Get File, Dir and Basename */
+			file = (uchar*)files.gl_pathv[i];
+			getBasename(basefilename, file);
+			getBasedir(basedir, file);
+
+			if(file[strlen((char*)file)-1] == '/')
+				continue;/* we cannot process subdirs! */
+
+			DBGPRINTF("fen_DirSearchFiles found matching file '%s' \n", file);
+
+			/* Get correct dirindex from basedir! */
+			dirIdx = dirsFindDir(basedir);
+			if (dirIdx == -1) {
+				/* Add dir to table and create watch */
+				CHKiRet(dirsAdd(basedir, &dirIdx));
+				DBGPRINTF("fen_DirSearchFiles adding new dir '%s' to dirs table idx %d\n", basedir, dirIdx);
+//				fen_processEventDir(NULL, dirIdx, 0); /* Monitor child directory as well */
+			}
+
+			/* Search for file index here */
+			ftIdx = fileTableSearchNoWildcard(&dirs[dirIdx].active, (uchar*)basefilename);
+			if(ftIdx >= 0) {
+				DBGPRINTF("fen_DirSearchFiles file '%s' idx %d already being monitored ... \n", file, ftIdx);
+			} else {
+				DBGPRINTF("fen_DirSearchFiles setup new monitor for dynamic file '%s' \n", file);
+
+				/* duplicate listener firs */
+				pLstnNew = pLstn;
+				if(lstnDup(&pLstnNew, basefilename, dirs[dirIdx].dirName) != RS_RET_OK) {
+					DBGPRINTF("fen_DirSearchFiles failed to duplicate listener for '%s'\n",
+						pLstn->pszFileName);
+					ABORT_FINALIZE(RS_RET_IO_ERROR);
```

Comment:
original error code from lstnDup should be preserved and use here!

Reply:
lstnDup can only fail with RS_RET_OUT_OF_MEMORY and RS_RET_ERR at the moment. 
Also other calls of lstnDup look the same. I would consider changing this in another issue (refactoring the code?).

Reply:
ok, so let's do that - but isn't it a 5 minute job?